llama_index/vector_stores/google/generativeai/base.py

```diff
@@ -0,0 +1,231 @@
+import logging
+import uuid
+from typing import Any, Dict, List, Optional, Sequence, cast
+
+from llama_index.bridge.pydantic import BaseModel, Field, PrivateAttr  # type: ignore
+from llama_index.indices.service_context import ServiceContext
+from llama_index.schema import BaseNode, RelatedNodeInfo, TextNode
+from llama_index.vector_stores.types import (
+    BasePydanticVectorStore,
+    MetadataFilters,
+    VectorStoreQuery,
+    VectorStoreQueryResult,
+)
+
+_logger = logging.getLogger(__name__)
+_import_err_msg = "`google.generativeai` package not found, please run `pip install google-generativeai`"
+_default_doc_id = "default-doc"
+
+
+google_service_context = ServiceContext.from_defaults(
+    # Avoids instantiating OpenAI as the default model.
+    llm=None,
+    # Avoids instantiating HuggingFace as the default model.
+    embed_model=None,
+)
+
+
+class GoogleVectorStore(BasePydanticVectorStore):
+    stores_text: bool = True
+    is_embedding_query: bool = False
+
+    # This is not the Google's corpus name but an ID generated in the LlamaIndex
+    # world.
+    corpus_id: str = Field(frozen=True)
+
+    _client: Any = PrivateAttr()
+
+    def __init__(self, *, client: Any, **kwargs: Any):
+        try:
+            import google.ai.generativelanguage as genai
+        except ImportError:
+            raise ImportError(_import_err_msg)
+
+        super().__init__(**kwargs)
+
+        assert isinstance(client, genai.RetrieverServiceClient)
+        self._client = client
+
+    @classmethod
+    def from_corpus(cls, *, corpus_id: str) -> "GoogleVectorStore":
+        try:
+            import llama_index.vector_stores.google.generativeai.genai_extension as genaix
+        except ImportError:
+            raise ImportError(_import_err_msg)
+
+        _logger.debug(f"\n\nGoogleVectorStore.from_corpus(corpus_id={corpus_id})")
+
+        return cls(corpus_id=corpus_id, client=genaix.build_retriever())
+
+    @classmethod
+    def create_corpus(
+        cls, *, corpus_id: Optional[str] = None, display_name: Optional[str] = None
+    ) -> "GoogleVectorStore":
+        try:
+            import llama_index.vector_stores.google.generativeai.genai_extension as genaix
+        except ImportError:
+            raise ImportError(_import_err_msg)
+
+        _logger.debug(
+            f"\n\nGoogleVectorStore.create_corpus(new_corpus_id={corpus_id}, new_display_name={display_name})"
+        )
+
+        client = genaix.build_retriever()
```

Comment:
nit & optional: would it be clearer if we say `build_semantic_retriever()` here? Retriever sounded like the reserved keyword from llamaindex